indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -1640,19 +1648,31 @@ private boolean isTaskCurrent(int taskGroupId, String taskId)
     SeekableStreamIndexTask<PartitionIdType, SequenceOffsetType> task = (SeekableStreamIndexTask<PartitionIdType, SequenceOffsetType>) taskOptional
         .get();
 
-    String taskSequenceName = task.getIOConfig().getBaseSequenceName();
+    String taskSequenceName = generateSequenceName(
```

Comment:
Can you add a comment explaining what is going on here and why this is doing this?

Reply:
Done